### PR TITLE
fix(queue): Ensure that after stages run with an authenticated context

### PR DESCRIPTION
**Pull request by ajordens**
This addresses a situation where the `DeployCanaryStage` failed to
plan after stages that involved a lookup against a restricted account.

### CODE REVIEW

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt [hunk 1]
```diff
@@ -78,7 +82,13 @@ class CompleteStageHandler(
               status = SKIPPED
             }
           } else if (status.isFailure) {
-            if (stage.planOnFailureStages()) {
+            var hasOnFailureStages = false
+
+            stage.withAuth {
+              hasOnFailureStages = stage.planOnFailureStages()
```
Comment (ajordens) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: I got some kotlin error when I nested all this within `stage.withAuth` ... something about `return@withStage` not being legitimate.

Broke this apart to get around it.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt [hunk 2]
```diff
@@ -64,7 +66,9 @@ class CompleteStageHandler(
             // check to see if this stage has any unplanned synthetic after stages
             var afterStages = stage.firstAfterStages()
             if (afterStages.isEmpty()) {
-              stage.planAfterStages()
+              stage.withAuth {
```
Comment (ajordens) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: `StartStageHandler` was already doing this which is why `DeployCanaryStage` half-worked.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt (hunk 1 above)
Comment (robfletcher) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: I always think that `return@something` syntax is a good sign that things need breaking apart so I'm 100% down with that.

Comment (marchello2000) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: reading through https://kotlinlang.org/docs/reference/returns.html and not understanding why you can't just do `return@withStage` here...  @robfletcher, if you know, would love to understand it.

**Pull request merged**